### PR TITLE
add Download Forms page [feature flagged]

### PR DESCRIPTION
**Pull request by nickpell**
@NoahCarnahan cc: @esoergel 

screen shot 2015-06-25 at 1 23 14 pm

### CODE REVIEW

File: corehq/apps/reports/dispatcher.py [hunk 1]
```diff
@@ -283,3 +283,10 @@ class DataExportInterfaceDispatcher(ProjectReportDispatcher):
 
     def dispatch(self, request, *args, **kwargs):
         return super(DataExportInterfaceDispatcher, self).dispatch(request, *args, **kwargs)
+
+class DataDownloadInterfaceDispatcher(ProjectReportDispatcher):
```
Comment (dimagimon) [inline corehq/apps/reports/dispatcher.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/reports/static/reports/javascripts/reports.async.js [hunk 2]
```diff
@@ -80,7 +81,9 @@ var HQAsyncReport = function (o) {
         }
         self.filterForm.submit(function () {
             var params = urlSerialize(this);
-            History.pushState(null,window.location.title,window.location.pathname+"?"+params);
+            History.pushState(null,window.location.title,
+                    window.location.pathname + '?' + params
+                    + (self.additionalWindowParams ? '&' + self.additionalWindowParams: ''));
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.async.js]: Bad line breaking before '+'.

File: corehq/apps/reports/static/reports/javascripts/reports.async.js [hunk 3]
```diff
@@ -80,7 +81,9 @@ var HQAsyncReport = function (o) {
         }
         self.filterForm.submit(function () {
             var params = urlSerialize(this);
-            History.pushState(null,window.location.title,window.location.pathname+"?"+params);
+            History.pushState(null,window.location.title,
+                window.location.pathname + '?' + params
+                + (self.additionalWindowParams ? '&' + self.additionalWindowParams: ''));
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.async.js]: Bad line breaking before '+'.

File: corehq/apps/reports/templates/reports/partials/download_export.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{% load i18n %}
+{% load hq_shared_tags %}
```
Comment (czue) [inline corehq/apps/reports/templates/reports/partials/download_export.html]: assume nothing obviously recyclable from other places in this file?

File: corehq/apps/reports/standard/export.py [hunk 5]
```diff
@@ -408,4 +411,31 @@ def saved_exports(self):
...
+                + '?export_id=' + export._id
+            )
         return sorted(exports, key=lambda x: x.name)
+
+
+class FormExportReport(FormExportReportBase):
+    base_template = 'reports/standard/export_download.html'
+    report_template_path = 'reports/partials/download_export.html'
+    name = ugettext_noop('Download Forms')
+    section_name = "Export Data"
```
Comment (czue) [inline corehq/apps/reports/standard/export.py]: intentionally not translated?

**Comment by czue** [comment]
(didn't review closely + no blockers, just random quick comments)

File: corehq/apps/reports/templates/reports/partials/download_export.html (hunk 4 above)
Comment (nickpell) [inline corehq/apps/reports/templates/reports/partials/download_export.html]: it was easier to pull out the code I needed, and when this goes live I'll delete the other code

File: corehq/apps/reports/standard/export.py (hunk 5 above)
Comment (nickpell) [inline corehq/apps/reports/standard/export.py]: nope, my mistake (fixing)

**Pull request merged**